src/components/account/FaucetPanel.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { useCtx } from "../../GlobalCtx";
+
+namespace FaucetStatus {
+  export function idle(): FaucetStatus {
+    return { state: FaucetState.Idle };
+  }
+  export function loading(): FaucetStatus {
+    return { state: FaucetState.Loading };
+  }
+  export function success(txHash: string): FaucetStatus {
+    return { state: FaucetState.Success, txHash };
+  }
+  export function error(message: string): FaucetStatus {
+    return { state: FaucetState.Error, message };
+  }
+}
+
+enum FaucetState {
+  Idle = "idle",
+  Loading = "loading",
+  Success = "success",
+  Error = "error",
+}
+
+type FaucetStatus =
+  | { state: FaucetState.Idle }
+  | { state: FaucetState.Loading }
+  | { state: FaucetState.Success; txHash: string }
+  | { state: FaucetState.Error; message: string };
+
+interface FaucetPanelProps {
+  selectedAddress: string;
+  onSuccess: (txHash: string) => void;
+}
+
+export function FaucetPanel({ selectedAddress, onSuccess }: FaucetPanelProps) {
+  const { collatorWsApi: api } = useCtx();
+  const [faucetStatus, setFaucetStatus] = useState<FaucetStatus>(FaucetStatus.idle);
+
+  const handleDrip = async () => {
+    if (!api || !selectedAddress) return;
+
+    try {
+      setFaucetStatus(FaucetStatus.loading);
+
+      const unsub = await api.tx.faucet.drip(selectedAddress).send((result) => {
+        const { status, dispatchError } = result;
+
+        if (status.isInBlock || status.isFinalized) {
+          if (dispatchError) {
+            if (dispatchError.isModule) {
+              const decoded = api.registry.findMetaError(dispatchError.asModule);
+              const { docs, name, section } = decoded;
+
+              const userMessage =
+                section === "faucet" && name === "FaucetUsedRecently"
+                  ? "You can only request tokens once every 24 hours."
+                  : docs.join(" ") || "Transaction failed.";
+
+              setFaucetStatus(FaucetStatus.error(userMessage));
+              setTimeout(() => setFaucetStatus(FaucetStatus.idle), 4000);
+            } else {
+              setFaucetStatus(FaucetStatus.error(dispatchError.toString()));
+              setTimeout(() => setFaucetStatus(FaucetStatus.idle), 4000);
+            }
+          } else {
+            const txHash = result.txHash.toHex();
+            setFaucetStatus(FaucetStatus.success(txHash));
+            onSuccess(txHash);
+          }
+
+          unsub();
+        }
+      });
+    } catch (err) {
+      const rawMessage = (err as Error).message || "";
```

Comment:
Mega sus, you can't just cast things like this without a good cause + You should at least have a fallback error message with logs too, something like "failed to decode error message"
```suggestion
	if (err instanceof Error) {
		...
	}
```